src/main/java/Duke.java

```diff
@@ -1,61 +1,93 @@
+        }
+        return commandCodeMapping.get("invalid command");
+    }
+
+    /**
+     * Switching logic for different commands.
+     * @param command user command
+     */
+    private static void commandProcessor(String command) {
+        Integer commandCode = getCommandCode(command);
+        if (commandCode == commandCodeMapping.get("list")) {
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.printf("%s[%s][%c] %s\n",
+                        i, tasks.get(i).getShortName(),
+                        (char) (Integer.parseInt(tasks.get(i).getStatusIcon(), 16)),
+                        tasks.get(i).description);
             }
-            break;
-        default:
-            Task t;
-            String taskType = command.split(" ", 2)[0];
-            String content = command.split(" ", 2)[1];
-            if (taskType.equals("todo")) {
-                t = new Todo(content);
-            } else if (taskType.equals("deadline")) {
-                String desc = content.split("/by")[0];
-                String deadlineTime = content.split("/by")[1];
-                t = new Deadline(desc, deadlineTime);
-            } else if (taskType.equals("event")) {
-                String desc = content.split("/at")[0];
-                String eventTime = content.split("/at")[1];
-                t = new Event(desc, eventTime);
-            } else {
+        } else if (commandCode == commandCodeMapping.get("mark done")) {
+            Integer taskIndex = Integer.parseInt(command.split(" ")[1]);
+            if (taskIndex < tasks.size()) {
+                tasks.get(taskIndex).markDone();
                 return;
             }
-            tasks.add(t);
-            System.out.println("Got it. I've added this task: " + command);
-            System.out.printf("Now you have %s tasks in this list\n", tasks.size());
+        } else if (commandCode == commandCodeMapping.get("todo")) {
+            String todoContent = command.split(" ", 2)[1];
+            Task todo = new Todo(todoContent);
+            tasks.add(todo);
+        } else if (commandCode == commandCodeMapping.get("deadline")) {
+            String deadlineContent = command.split(" ", 2)[1];
+            String deadlineDesc = deadlineContent.split("/by")[0];
+            String deadlineTime = deadlineContent.split("/by")[1];
+            Task deadline = new Deadline(deadlineDesc, deadlineTime);
+            tasks.add(deadline);
+        } else if (commandCode == commandCodeMapping.get("event")) {
+            String eventContent = command.split(" ", 2)[1];
+            String eventDesc = eventContent.split("/at")[0];
+            String eventTime = eventContent.split("/at")[1];
+            Task event = new Event(eventDesc, eventTime);
+            tasks.add(event);
+        } else if (commandCode == commandCodeMapping.get("invalid command")) {
+            System.out.printf("OOPS Wrong command\n");
         }
```

Comment:
What if there is any case that passes all your if-statement?

Reply:
CommandParser will be in charge of command validation and categorization. so any command will get a corresponding commandCode and it will definitely be one of the if-statement.

Reply:
I see, however, still, the logic looks unstable. Although the commandcodeMapping instance won't miss any default case, you could add else statement here to clarify the flow of this if-else statement.